processing/src/main/java/org/apache/druid/segment/nested/NestedDataColumnSerializer.java

```diff
@@ -374,11 +375,17 @@ public void serializeDictionaries(
 
   @Override
   public void serialize(ColumnValueSelector<? extends StructuredData> selector) throws IOException
+  {
+    serialize(StructuredData.wrap(selector.getObject()));
+  }
+
+  public void serialize(StructuredData data) throws IOException
   {
     if (!dictionarySerialized) {
-      throw new ISE("Must serialize value dictionaries before serializing values for column [%s]", name);
+      throw DruidException.defensive(
+          "Must serialize value dictionaries before serializing values for column [%s]", name
+      );
     }
-    StructuredData data = StructuredData.wrap(selector.getObject());
```

Comment:
The order is changed a bit. 
Earlier, exception was checked first and then the wrapping was done. Is that a problem ?

Reply:
StructuredData.wrap is a simple constructor, so I think this is fine